process/guardian/guardedAccount.go

```diff
@@ -244,10 +244,7 @@ func (agc *guardedAccount) saveAccountGuardians(account vmcommon.UserAccountHand
 
 func (agc *guardedAccount) getConfiguredGuardians(uah state.UserAccountHandler) (*guardians.Guardians, error) {
 	guardiansMarshalled, err := uah.RetrieveValueFromDataTrieTracker(guardianKey)
-	if err != nil {
-		return nil, err
-	}
-	if len(guardiansMarshalled) == 0 {
+	if err != nil || len(guardiansMarshalled) == 0 {
```

Comment:
maybe return &guardians.Guardians{Slice: make([]*guardians.Guardian, 0)} also if Unmarshal returns error. You may also remove the error from return type

Reply:
for unmarshall error I would propagate the error, as it would be an issue with the save and it needs to be detected & fixed.